dev/fattest.simplicity/src/componenttest/custom/junit/runner/TestModeFilter.java

```diff
@@ -73,6 +73,17 @@ public boolean shouldRun(Description desc) {
             testMode = mode.value();
         }
 
+        boolean run = shouldRun(testMode);
+
+        if (!run) {
+            Log.debug(getClass(), "Removing test " + desc.getMethodName() + " with mode " + testMode + " from list to run, because not valid for current mode "
+                                  + FRAMEWORK_TEST_MODE);
+            return false;
+        } else
+            return true;
+    }
+
+    public static boolean shouldRun(TestMode testMode) {
```

Comment:
Not sure that this change is actually an improvement

Reply:
it isn't except that i can call the new method from my code